test/nuxt/composables.test.ts

```diff
@@ -691,6 +691,26 @@ describe('useCookie', () => {
     expect(computedVal.value).toBe(0)
   })
 
+  it('cookie decode function should be invoked once', () => {
```

Comment:
this test passes on main branch as well - would you look at this? 🙏

Reply:
> this test passes on main branch as well - would you look at this? 🙏This test passes on main branch as well - would you look at this? 🙏

After looking at your refactoring and simplification of my code, I found no problems. Is there anything else I need to cooperate with?

Reply:
the code is good - but the test doesn't test it, as it also passes on the `main` branch, even without this PR. would you update the test so it _fails_ on the main branch, but passes with this PR? 🙏

Reply:
OK, I found that the new version of cookie-es does not have the code I changed. I will ask what is happening with pi0, and then add test cases.

Reply:
Cookie filter test case added.
by the way, the judgment conditions of the filter have been revised.